lib/modules/datasource/rubygems/versions-datasource.ts

```diff
@@ -49,132 +90,209 @@ const Lines = z
     }
 
-    const versions = regCache.packageReleases.get(packageName);
-    if (!versions) {
+    const packageVersions = cache.packageVersions.get(packageName);
+    if (!packageVersions?.length) {
+      logger.debug(
+        { packageName, registryUrl },
+        'Rubygems: versions not found'
+      );
       return null;
     }
 
-    const releases = versions.map((version) => ({ version }));
+    const releases = packageVersions.map((version) => ({ version }));
     return { releases };
   }
 
-  private updatePackageReleases(
-    packageReleases: PackageReleases,
-    lines: Lines
-  ): void {
-    for (const line of lines) {
-      const packageName = copystr(line.packageName);
-      let versions = packageReleases.get(packageName) ?? [];
-
-      const { deletedVersions, addedVersions } = line;
-
-      if (deletedVersions.size > 0) {
-        versions = versions.filter((v) => !deletedVersions.has(v));
-      }
-
-      if (addedVersions.length > 0) {
-        const existingVersions = new Set(versions);
-        for (const addedVersion of addedVersions) {
-          if (!existingVersions.has(addedVersion)) {
-            const version = copystr(addedVersion);
-            versions.push(version);
-          }
-        }
+  async fullSync(registryUrl: string): Promise<VersionsEndpointCache> {
+    try {
+      const url = `${registryUrl}/versions`;
+      const opts: HttpOptions = { headers: { 'Accept-Encoding': 'gzip' } };
+      const { body } = await this.http.get(url, opts);
+      return parseFullBody(body);
+    } catch (err) {
+      if (err instanceof HttpError && err.response?.statusCode === 404) {
+        return { versionsEndpointSupported: false };
       }
 
-      packageReleases.set(packageName, versions);
+      this.handleGenericErrors(err);
     }
   }
 
-  async updateRubyGemsVersions(regCache: RegistryCache): Promise<void> {
-    const url = `${regCache.registryUrl}/versions`;
-    const options = {
-      headers: {
-        'accept-encoding': 'gzip',
-        range: `bytes=${regCache.contentLength}-`,
-      },
-    };
-    let newLines: string;
+  async deltaSync(
+    oldCache: VersionsEndpointData,
+    registryUrl: string
+  ): Promise<VersionsEndpointCache> {
     try {
-      logger.debug('Rubygems: Fetching rubygems.org versions');
-      const startTime = Date.now();
-      newLines = (await this.http.get(url, options)).body;
-      const durationMs = Math.round(Date.now() - startTime);
-      logger.debug(`Rubygems: Fetched rubygems.org versions in ${durationMs}`);
-    } catch (err) /* istanbul ignore next */ {
-      if (err instanceof HttpError && err.response?.statusCode === 404) {
-        regCache.isSupported = false;
-        return;
-      }
+      const url = `${registryUrl}/versions`;
+      const startByte = oldCache.contentLength - oldCache.contentTail.length;
+      const opts: HttpOptions = {
+        headers: {
+          ['Accept-Encoding']: 'deflate, compress, br',
```

Comment:
Any benefit to adding `gzip` here too?

Reply:
Our http client will throw low-level error, something related to `zlib` buffer